eth2util/deposit/deposit.go

```diff
@@ -170,35 +150,22 @@ func withdrawalCredsFromAddr(addr string) ([32]byte, error) {
 		return [32]byte{}, errors.New("invalid withdrawal address", z.Str("address", addr))
 	}
 
-	var creds []byte
-
-	// Append the single byte ETH1_ADDRESS_WITHDRAWAL_PREFIX as prefix.
-	creds = append(creds, eth1AddressWithdrawalPrefix)
-
-	// Append 11 bytes of 0.
-	creds = append(creds, zeroBytes11...)
-
-	addr = strings.TrimPrefix(addr, "0x")
-	addrBytes, err := hex.DecodeString(addr)
+	addrBytes, err := hex.DecodeString(strings.TrimPrefix(addr, "0x"))
 	if err != nil {
 		return [32]byte{}, errors.Wrap(err, "decode address")
 	}
 
-	// Finally, append 20 bytes of ethereum address.
-	creds = append(creds, addrBytes...)
-
-	var resp [32]byte
-	copy(resp[:], creds)
+	var creds [32]byte
+	copy(creds[0:], eth1AddressWithdrawalPrefix) // Add 1 byte prefix.
+	copy(creds[12:], addrBytes)                  // Add 20 bytes of ethereum address suffix.
```

Comment:
more succinct to work directly with array, then zero11 also not required.

Reply:
yes, since the default zero value of a byte is `0`